cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -176,6 +177,13 @@ func (c Client) AddMember(name string, peerAddrs string) ([]Member, error) {
 		}
 	}
 
+	// Add the new member client address to the list of endpoints
+	parsedPeerAddrs, err := url.Parse(peerAddrs)
```

Comment:
If we move only the `url.Parse` + `if err != nil`  at the beginning of the function we can address @rosti comment "The call to url.Parse (in theory) can fail after successful call to MemberAdd, thus failing the AddMember while we actually have a new member." while preserving your working solution.
wdyt?